tests/templating.env

```diff
@@ -1,2 +1,13 @@
-_INCLUDE_BASIC=basic.env
-NAME=envy-{{VERSION}}
\ No newline at end of file
+VERSION=1.0.0
+NAME=envy
+DOUBLE={{NAME}}-{{VERSION}}-pro
+DOUBLE_NO_CLOSE={{NAME}}-{{VERSION-pro
+DOUBLE_OPEN={{NAME-{{VERSION}}-pro
+NOT_TEMPLATE=envy-{{}}-pro
+NO_CLOSE_START={{-envy
+NO_CLOSE_MIDDLE=envy-{{-pro
+NO_CLOSE_END=envy-{{
+VAR_START={{VERSION}}-pro
+VAR_MIDDLE=envy-{{VERSION}}-pro
```

Comment:
Will something like the below pass and substitute version as expected?

VAR_START={{{VERSION}}-pro}
VAR_MIDDLE=envy-{{{VERSION}}-pro}
VAR_END=envy-{{{VERSION}}}

Reply:
This all result in an error of `Error while parsing tempate, {VERSION not set` which is correct.
If you replace it with the correct version tag but keep the final } it is also ok
```
VAR_START={{VERSION}}-pro}
VAR_MIDDLE=envy-{{VERSION}}-pro}
VAR_END=envy-{{VERSION}}}
```
->
```
export VAR_END=envy-1.0.0}
export VAR_MIDDLE=envy-1.0.0-pro}
export VAR_START=1.0.0-pro}
```

Reply:
I guess we read code differently. I'd have expected the below by default

```
VAR_START={{{VERSION}}-pro}
VAR_MIDDLE=envy-{{{VERSION}}-pro}
VAR_END=envy-{{{VERSION}}}
```
->
```
export VAR_END={envy-1.0.0}
export VAR_MIDDLE={envy-1.0.0-pro}
export VAR_START={1.0.0-pro}
```
I'm not too fussed, edge case :)